### DISCUSSION TITLE
Expanding / recursing multi-level `all_columns` or `all_related`

### DISCUSSION THREAD

**Discussion by gregglind** [piccolo-orm/piccolo#634] [Ideas]
Suppose a request to 'show all information about all concert'.

Currently I might query like this:
```
await Concert.select(
    Concert.all_columns(), 
    Concert.band_1.all_columns(), 
    Concert.band_2.all_columns(), 
    Concert.band_1.manager.all_columns(), 
    Concert.band_2.manager.all_columns(), 
    Concert.venue.all_columns()
).output(nested=True)
```

1/ Is there a shorter idiom for recursive, or multi-level expansion from foreignKeys?  
2/ If not, should there be?   

Editorial:
- Use case:  recursive, nested expansion fits a lot of display idioms on the web and reporting.

**Comment by dantownsend** [comment]
I can see how it gets a bit verbose for tables with lots of foreign keys, and which are highly nested.

We could maybe add something like this?

```python
await Concert.select(Concert.walk(depth=1))
```

I'm open to other ideas though.

**Comment by gregglind** [comment]
Thanks for the quick response.  

1/. `.walk(depth=int)` is great!  Alternatives I had concidered:  `expand(levels)`.  Not sure which is more Pythonic, or `orm`ish.
2/ Transformations into non-tabular, nested data is a place where ORMS beat raw SQL, and where GraphQL shines.    Article +  comments + commenter slugs.
3/ `walk` supports awesome M2M affordances as well, another place where ORMs beat raw SQL.
